shiny/reactive/_reactives.py

```diff
@@ -531,7 +531,16 @@ async def _run(self) -> None:
         with session_context(self._session):
             try:
                 with ctx():
-                    await self._fn()
+                    res = await self._fn()
+
+                    if res is not None:
+                        warnings.warn(
+                            "reactive.Effect function returned a value, but should return None; "
+                            "reactive.Effects should only be used for side effects. "
+                            "Did you mean to use reactive.Calc instead?",
+                            ReactiveWarning,
+                            stacklevel=2,
+                        )
```

Comment:
I like this PR--can you just tell me if the warning here gives any helpful indication as to which reactive.Effect is causing the problem? If not, we could use `inspect.unwrap(foobar._fn).__code__.co_filename` and `.co_firstlineno`

Reply:
IIRC, the problem is that we don't know how many times the user-defined function has been wrapped, so we don't know how many levels in the stack we need to skip.

For example, if the user function is synchronous, then we wrap it once in an async function. But they may decorated it with `@reactive.event`, which would add another layer.

Now, if the user function threw an exception, I think we would be able to catch that and find the source of it, but that's not the situation here.